crates/sui-aws-orchestrator/README.md

```diff
@@ -0,0 +1,93 @@
+# Orchestrator
+
+The Orchestrator crate provides facilities for quickly deploying and benchmarking this codebase in a geo-distributed environment. Please note that it is not intended for production deployments or as an indicator of production engineering best practices. Its purpose is to facilitate research projects by allowing benchmarking of (variants of) the codebase and analyzing performance.
+
+This guide provides a step-by-step explanation of how to run geo-distributed benchmarks on [Amazon Web Services (AWS)](http://aws.amazon.com).
+
+## Step 1. Set up cloud provider credentials
+
+To enable programmatic access to your cloud provider account from your local machine, you need to set up your cloud provider credentials. These credentials authorize your machine to create, delete, and edit instances programmatically on your account.
+
+### Setting up AWS credentials
+
+1. Find your ['access key id' and 'secret access key'](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html#cli-configure-quickstart-creds).
+2. Create a file `~/.aws/credentials` with the following content:
+
+```text
+[default]
+aws_access_key_id = YOUR_ACCESS_KEY_ID
+aws_secret_access_key = YOUR_SECRET_ACCESS_KEY
+```
+
+Do not specify any AWS region in that file, as the scripts need to handle multiple regions programmatically.
+
+## Step 2. Specify the testbed configuration
+
+Create a file called `settings.json` that contains all the configuration parameters for the testbed deployment. You can find an example file at `./assets/settings.json` with the following content:
+
+```json
+{
+  "testbed_id": "alberto",
+  "cloud_provider": "aws",
+  "token_file": "/Users/alberto/.aws/credentials",
+  "ssh_private_key_file": "/Users/alberto/.ssh/aws",
+  "regions": [
+    "us-east-1",
+    "us-west-2",
+    "ca-central-1",
+    "eu-central-1",
+    "ap-northeast-1",
+    "eu-west-1",
+    "eu-west-2",
+    "ap-south-1",
+    "ap-southeast-1",
+    "ap-southeast-2"
+  ],
+  "specs": "m5d.8xlarge",
+  "repository": {
+    "url": "https://github.com/MystenLabs/sui.git",
+    "commit": "orchestrator"
+  },
+  "results_directory": "./results",
+  "logs_directory": "./logs"
+}
+```
+
+The documentation of the `Settings` struct in `./src/settings.rs` provides detailed information about each field and indicates which ones are optional. If you're working with a private GitHub repository, you can include a [private access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens) in the repository URL. For example, if your access token is `[your_token]`, the repository URL should be formatted as follows:
+
+```json
+"repository": {
+  "url": "http://[your_token]@github.com/mystenlabs/sui.git",
+  "commit": "orchestrator"
```

Comment:
What does this field do?

Reply:
This field allows you to deploy a specific branch/commit of the repo to the nodes. It is particularly handy when you work on your current branch and want to quickly test your changes.

Reply:
I wonder if the orchestrator branch / tag will always exist in Sui. Maybe just use `main` before merging?